test_assets/PrimvarInterpolation/README.md

```diff
@@ -0,0 +1,221 @@
+
+This example asset demonstrates each primvar interpolation type. The example was inspired by / plagiarized from a [usd-interest post by Jeremy Cowles](https://groups.google.com/g/usd-interest/c/MqYLLkp-rYA). 
+
+# Mesh construction
+
+It will be helpful to review how mesh geometry is defined in USD. For this example, we define a mesh with two faces:
+
+```
+def Mesh "example"
+{
+    float3[] extent = [(-1, 0, 0), (1, 1, 0)]
+    point3f[] points = [(-1, 0, 0), (-1, 1, 0), (0, 1, 0), (0, 0, 0), (1, 1, 0), (1, 0, 0)]
+    int[] faceVertexCounts = [4, 4]
+    int[] faceVertexIndices = [3, 2, 1, 0, 5, 4, 2, 3]
+}
+```
+
+![example mesh](images/example_mesh.png)
+
+The mesh is constructed from six vertices, which are defined using the **points** attribute:
+
+```
+point3f[] points = [(-1, 0, 0), (-1, 1, 0), (0, 1, 0), (0, 0, 0), (1, 1, 0), (1, 0, 0)]
+```
+
+This image shows the points and their indices within the **points** array:
+
+![mesh points and indices](images/points_and_ids.png)
+
+Once the points are defined, the **faceVertexCount** and **faceVertexIndices** attributes define how a mesh is constructed from those points. **faceVertexCount** is an ordered array of how many points / vertices are used to define each face of the mesh. **faceVertexIndices** is an ordered array of _which_ points / vertices comprise each face, using the index of the points as defined in the **points** attribute. 
+
+![faceVertexIndices](images/vertindices.png)
+
+The order of the **faceVertexIndices** array aligns with the order of the **faceVertexCounts** array. If the first face listed in the **faceVertexCounts** array has _four_ points, then the first _four_ entries **faceVertexIndices** array define the points that comprise the first face of the mesh. If we name our faces A and B, these attributes define A and B in this order:
+
+```
+int[] faceVertexCounts = [4, 4]
+# int[] faceVertexCounts = [A, B]
+int[] faceVertexIndices = [3, 2, 1, 0, 5, 4, 2, 3]
+# int[] faceVertexIndices = [A_0, A_1, A_2, A_3, B_0, B_1, B_2, B_3]
+```
+
+![faces defined from faceVertexIndices](images/faces_defined.png)
+
+# Primvar interpolation with Mesh prims
+
+For these examples, we'll use the displayColor primvar, because Storm and most other render delegates can display it natively.
+
+The example file [primvar_interpolation.usda](primvar_interpolation.usda) was used to generated the screenshot in the overview of this document, and contains all of the examples discussed below.
+
+## constant
+
+![constant interpolation](images/constant.png)
+
+```
+color3f[] primvars:displayColor = [(1, 0.3, 0.3)] (
+    interpolation = "constant"
+)
+```
+
+**Pixar's definition**
+> One value remains constant over the entire surface primitive.
+
+**Size of primvar array**
+
+1
+
+**Notes**
+
+With constant interpolation, a single value is specified for the entire prim. It is important to note, this is still specified as an array, even though it will only have a single entry.
```

Comment:
I don't know if you should make this super explicit, but it might be worth mentioning once:

The primvar is declared as color3f[] (which is a tuple of three values), so a single constant primvar value is an array of one tuple of three values.

```color3f[] primvars:displayColor = [(1, 0.3, 0.3)]```